docs/how-chugsplash-works.md

```diff
@@ -0,0 +1,252 @@
+
+Each of these functions exists on the `ChugSplashManager`. If the executor attempts to send a transaction that isn't in the correct order, the call will revert.
+
+The reason for steps 1 and 3 is to ensure that the deployment or upgrade is **atomic**. This is less important for fresh deployments, but it's critical for upgrades because it ensures end-users aren't interacting with a partially initialized set of smart contracts while the upgrade is occurring.
+
+For every `SetStorage` and `DeployImplementation` action, the executor must submit a corresponding Merkle proof. The root of this Merkle tree was originally supplied by the user in the `propose` step.
+
+You might be wondering how the `SetStorage` actions in step 2 can be executed in a standard proxy, especially if the proxy's implementation is set to `address(0)` in the first step. We achieve this by doing the following in each  `executeAction` transaction:
+
+1. Upgrade the proxy's implementation from `address(0)` to a `ProxyUpdater` contract, which contains just a single function:
+```sol
+function setStorage(bytes32 _key, bytes32 _value) external {
+    assembly {
+        sstore(_key, _value)
+    }
+}
+```
+2. Trigger the `setStorage` function which is now available on the proxy.
+3. Set the proxy's implementation back to `address(0)`.
+
+Since these steps all occur over the course of a *single* transaction, the proxy's implementation is never set to anything other than `address(0)` *between* transactions. This ensures a malicious attacker can't trigger a `setStorage` call on the proxy.
+
+## Wrapping Up
+
+Congrats, you made it to the end!
+
+If you have questions or comments, we'd love to hear from you in our [Discord](https://discord.com/invite/CqUPhgRrxq).
+
+## FAQ
+
+### Why does ChugSplash encode variables into storage slot values instead of just calling existing functions on the contract?
+
+This allows us to guarantee that your deployment is deterministic. We wouldn't be able to make this guarantee if we called functions because they can contain arbitrary logic. This means we wouldn't be able to reliably determine the end-state of the upgrade or even which state variables are changing.
+
+Our approach also removes the need for initializer functions, which removes the threat of uninitialized proxies getting hijacked by attackers. Uninitialized proxies have been the cause of some notable vulnerabilities, including the Nomad bridge hack, the $10 million bug bounty paid by Wormhole, and the $2 million bug bounty paid by Arbitrum.
+
+### If a multisig or governance is approving the deployment from a UI, and anybody could be executing it, how can you ensure that the contracts are verified on Etherscan?
+
+There are two solutions to this problem.
+1. You can execute your own deployment, which is as easy as spinning up a Docker container.
+2. You can sign up for the ChugSplash Managed Service (described [here](#wrapping-up)), which will verify the contracts on your behalf for free.
```

Comment:
You said previously that the managed service is invite-only. We should probably reconcile that with what you're saying here. 

I recommend just removing this faq entry for now. We can come back and update it to add more info about Etherscan verification once we've nailed down how the managed service is going to work.